src/Mesh/babylon.mesh.ts

```diff
@@ -1621,79 +1621,162 @@
+        }
+        // handle special case of extra verts.  (in theory gltf can handle 12 influences)
+        private normalizeSkinWeightsAndExtra(): void {
+            let matricesWeightsExtra = (<FloatArray>this.getVerticesData(VertexBuffer.MatricesWeightsExtraKind));
+            let matricesWeights = (<FloatArray>this.getVerticesData(VertexBuffer.MatricesWeightsKind));
+            let numWeights = matricesWeights.length;
+            for (var a=0; a<numWeights; a+=4){
+                var t = matricesWeights[a] + matricesWeights[a+1] +matricesWeights[a+2] +matricesWeights[a+3];
+                t += matricesWeightsExtra[a] + matricesWeightsExtra[a+1] +matricesWeightsExtra[a+2] +matricesWeightsExtra[a+3];
+                // check for invalid weight and just set it to 1.
+                if (t===0) matricesWeights[a] = 1;
+                else{
+                    // renormalize so everything adds to 1 use reciprical 
+                    let recip = 1 / t;
+                    matricesWeights[a]*=recip;
+                    matricesWeights[a+1]*=recip;
+                    matricesWeights[a+2]*=recip;
+                    matricesWeights[a+3]*=recip;
+                    // same goes for extras
+                    matricesWeightsExtra[a]*=recip;
+                    matricesWeightsExtra[a+1]*=recip;
+                    matricesWeightsExtra[a+2]*=recip;
+                    matricesWeightsExtra[a+3]*=recip;
                 }
+                
             }
+            this.setVerticesData(VertexBuffer.MatricesWeightsKind, matricesWeights);
+            this.setVerticesData(VertexBuffer.MatricesWeightsKind, matricesWeightsExtra);
+        }
 
-            this.setVerticesData(VertexBuffer.MatricesIndicesKind, matricesIndices);
-            if (matricesIndicesExtra) {
-                this.setVerticesData(VertexBuffer.MatricesIndicesExtraKind, matricesIndicesExtra);
+        /**
+         * ValidateSkinning is used to determin that a mesh has valid skinning data along with skin metrics, if missing weights, 
+         * or not normalized it is returned as invalid mesh the string can be used for console logs, or on screen messages to let
+         * the user know there was an issue with importing the mesh
+         * @returns a validation object with skinned, valid and report string
+         */
+        public validateSkinning() : {skinned:boolean, valid:boolean, report:string} {
+            let matricesWeightsExtra = (<FloatArray>this.getVerticesData(VertexBuffer.MatricesWeightsExtraKind));
+            let matricesWeights = (<FloatArray>this.getVerticesData(VertexBuffer.MatricesWeightsKind));
+            if (matricesWeights===null || this.skeleton==null)
+            {
+                return {skinned:false, valid: true, report:"not skinned"}
+            }
+
+            let numWeights = matricesWeights.length;
+            let numberNotSorted :number = 0;

```

Comment:
Type should be declared with Var f: type =
(the space is after the :)

We will add a tslinter soon to make sure to fix that automatically :)

Reply:
I also did a lint pass manually over the code to keep it consistent with the style over all functions. hopefully that looks cleaner